From 217705da7726002ffe61dad51a6c9cc97c52f649 Mon Sep 17 00:00:00 2001
From: Himanshu Agarwal <himanaga@codeaurora.org>
Date: Wed, 4 Oct 2017 14:59:52 +0530
Subject: qcacld-3.0: Add sanity check for vdev id to prevent OOB access

Add sanity check for vdev id in wma_scan_event_callback() to
prevent out of bound access of memory.

Change-Id: I2a264029a4d80a5478f786f91923ca26af2db74a
CRs-Fixed: 2119399
---
 core/wma/src/wma_scan_roam.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/core/wma/src/wma_scan_roam.c b/core/wma/src/wma_scan_roam.c
index 62f7028..967a091 100644
--- a/core/wma/src/wma_scan_roam.c
+++ b/core/wma/src/wma_scan_roam.c
@@ -5994,6 +5994,12 @@ int wma_scan_event_callback(WMA_HANDLE handle, uint8_t *data,
 
 	param_buf = (WMI_SCAN_EVENTID_param_tlvs *) data;
 	wmi_event = param_buf->fixed_param;
+
+	if (wmi_event->vdev_id >= wma_handle->max_bssid) {
+		WMA_LOGE("Invalid vdev id from firmware");
+		return -EINVAL;
+	}
+
 	vdev_id = wmi_event->vdev_id;
 	scan_id = wma_handle->interfaces[vdev_id].scan_info.scan_id;
 
-- 
cgit v1.1

